File: python/tests/docs/test_concept.py
from sp_svg_diagram import SVGDiagram

from .utils import compare_svg


def test_empty():
    diagram = SVGDiagram()
    svg = diagram.render()
    compare_svg("concept", "empty", svg)


def test_node():
    diagram = SVGDiagram()
    node = diagram.add_node("A")
    node.set_center(0, 0)
    node.set_label("foo\nbar")
    svg = diagram.render()
    compare_svg("concept", "node", svg)


def test_edge():
    diagram = SVGDiagram()
    node1 = diagram.add_node("A")
    node1.set_center(0, 0)
    node2 = diagram.add_node("B")
    node2.set_center(60, 80)
    diagram.add_edge("A", "B")
    svg = diagram.render()
    compare_svg("concept", "edge", svg)

File: tests/docs/test_concept.cpp
#include "svg_diagram.h"

#include <cmath>
#include <numbers>
#include <format>
#include <gtest/gtest.h>
using namespace std;
using namespace svg_diagram;

static const auto OUTPUT_DIR = string("../docs/source/_static/concept/");

TEST(TestDocsConcept, Empty) {
    SVGDiagram diagram;
    diagram.render(OUTPUT_DIR + "empty.svg");
}

TEST(TestDocsConcept, Node) {
    SVGDiagram diagram;
    const auto node = diagram.addNode("A");
    node->setCenter(0, 0);
    node->setLabel("foo\nbar");
    diagram.render(OUTPUT_DIR + "node.svg");
}

TEST(TestDocsConcept, Edge) {
    SVGDiagram diagram;
    const auto node1 = diagram.addNode("A");
    node1->setCenter(0, 0);
    const auto node2 = diagram.addNode("B");
    node2->setCenter(60, 80);
    diagram.addEdge("A", "B");
    diagram.render(OUTPUT_DIR + "edge.svg");
}